sdk/src/azure/platform/CMakeLists.txt

```diff
@@ -82,3 +82,58 @@ if (TRANSPORT_CURL)
   target_include_directories(az_curl INTERFACE ${CURL_INCLUDE_DIR})
 
 endif()
+
+# Mosquitto Platform
+
+if (TRANSPORT_MOSQUITTO AND AZ_PLATFORM_IMPL STREQUAL "POSIX" AND NOT APPLE) # Temporary
+  find_package(mosquitto  CONFIG)
+  if(NOT mosquitto_FOUND)
+    find_package(mosquitto REQUIRED)
+  endif()
+
+  find_package(OpenSSL CONFIG)
+  if(NOT OpenSSL_FOUND)
+    find_package(OpenSSL REQUIRED)
+  endif()
+
+  add_library (
+    az_mqtt_mosquitto
+      STATIC
+      ${CMAKE_CURRENT_LIST_DIR}/az_mosquitto.c
+      ${CMAKE_CURRENT_LIST_DIR}/az_mqtt_policy.c
+  )
+
+  add_compile_definitions(az_mqtt_mosquitto
+    LIBMOSQUITTO_STATIC
+  )
+
+  target_link_libraries(az_mqtt_mosquitto PRIVATE az_core)
+
+  target_link_libraries(az_mqtt_mosquitto 
+    PUBLIC 
+      ${MOSQUITTO_LIBRARIES}
+      ${PTHREAD_LIBRARIES}
+      #${MOSQUITTO_LIBRARY}
+      OpenSSL::SSL 
+      OpenSSL::Crypto
+  )
+
+  target_include_directories(az_mqtt_mosquitto INTERFACE ${MOSQUITTO_INCLUDE_DIR})
+
+  # make sure that users can consume the project as a library.
+  add_library (az::mqtt_impl ALIAS az_mqtt_mosquitto)
+endif() # TRANSPORT_MOSQUITTO
+
+# No platform
+
+if (AZ_MQTT_TRANSPORT_IMPL STREQUAL "NONE" AND NOT APPLE)
```

Comment:
Does there need to be a fallthrough case for apple?

Reply:
We don't want to build any of the `az_mqtt` files for apple (even `az_nomqtt`) since they depend on `az_platform` which we have not implemented for it. This made me think that I should replace the `NOT APPLE` conditional with `(UNIX AND NOT APPLE)` since we only want to build this for linux